Lib/test/test_asyncio/test_taskgroups.py

```diff
@@ -741,6 +741,35 @@ async def coro2(g):
 
         self.assertEqual(get_error_types(cm.exception), {ZeroDivisionError})
 
+    async def test_taskgroup_context_manager_exit_raises(self):
+        # See https://github.com/python/cpython/issues/95289
+        class CustomException(Exception):
+            pass
+
+        async def raise_exc():
+            raise CustomException
+
+        @contextlib.asynccontextmanager
+        async def database():
+            try:
+                yield
+            finally:
+                raise CustomException
+
+        async def main():
+            task = asyncio.current_task()
+            try:
+                async with taskgroups.TaskGroup() as tg:
+                    async with database():
+                        tg.create_task(raise_exc())
+                        await asyncio.sleep(1)
+            except* CustomException:
+                self.assertEqual(task.cancelling(), 0)
```

Comment:
I find this test/example a bit confusing. It's clear that `CustomException` is raised but it's less clear *which* `CustomException` is being caught: the original one, raised by `raise_exc()`, or the one raised by the context manager. The answer is both -- and to verify that, we could add this:
```suggestion
            try:
                async with taskgroups.TaskGroup() as tg:
                    async with database():
                        tg.create_task(raise_exc())
                        await asyncio.sleep(1)
            except* CustomException as err:
                self.assertEqual(task.cancelling(), 0)
                self.assertEqual(len(err.exceptions), 2)

```
(One instance is the `raise_exc()` task, the other is the parent task being cancelled in the sleep.)